source/ui/BioXAS/BioXASDBHRMirrorView.cpp

```diff
@@ -0,0 +1,74 @@
+#include "BioXASDBHRMirrorView.h"
+
+BioXASDBHRMirrorView::BioXASDBHRMirrorView(BioXASDBHRMirror *mirror, QWidget *parent) :
+    QWidget(parent)
+{
+	// Initialize member variables.
+
+	mirror_ = 0;
+
+	layout_ = new QVBoxLayout();
+	setLayout(layout_);
+
+	pitchEditor_ = 0;
+	m1Editor_ = 0;
+	m2Editor_ = 0;
+
+	// Current settings.
+
+	setMirror(mirror);
+}
+
+BioXASDBHRMirrorView::~BioXASDBHRMirrorView()
+{
+
+}
+
+void BioXASDBHRMirrorView::setMirror(BioXASDBHRMirror *newMirror)
+{
+	if (mirror_ != newMirror) {
+
+		if (mirror_) {
+
+			// remove mirror editors from view layout.
+
+			layout_->removeWidget(pitchEditor_);
+			layout_->removeWidget(m1Editor_);
+			layout_->removeWidget(m2Editor_);
+
+			// delete mirror editors.
+
+			pitchEditor_->deleteLater();
+			pitchEditor_ = 0;
+
+			m1Editor_->deleteLater();
+			m1Editor_ = 0;
+
+			m2Editor_->deleteLater();
+			m2Editor_ = 0;
+		}
+
+		mirror_ = newMirror;
+
+		if (mirror_) {
+
+			// create mirror editors, add to view layout.
+
+			pitchEditor_ = new AMExtendedControlEditor(mirror_->pitchControl());
```

Comment:
Hey @helfrij, since you've had to implement methods like this a couple times over the past few tasks, how would you feel about adding the AMControlEditor by having a ```setControl()``` method?  Since you've been having to remove and delete the widgets and then making them again seems like a bit of a waste.  Maybe make a new issue for adding to the functionality and then doing some code cleanup on these?